Hestia/app/src/main/java/com/rugged/application/hestia/PeripheralFragment.java

```diff
@@ -0,0 +1,105 @@
+package com.rugged.application.hestia;
+
+import android.os.AsyncTask;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.EditText;
+
+import java.util.UUID;
+
+/**
+ * Created by Mark on 4-3-2017.
+ */
+
+//Fragment class taking care of a specific Peripheral
+public class PeripheralFragment extends Fragment {
+    private Peripheral mPeripheral;
+    //lock and unlock button
+    private Button onButton;
+    private Button offButton;
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        //receive the id from the PeripheralActivity
+        int peripheralId = (int) getActivity().getIntent()
+                .getSerializableExtra(PeripheralActivity.EXTRA_PERIPHERAl_ID);
+        mPeripheral = PeripheralLab.get(getActivity()).getPeripheral(peripheralId);
+
+
+    }
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View v = inflater.inflate(R.layout.fragment_peripheral, container, false);
+        onButton = (Button)v.findViewById(R.id.on_button);
+        offButton = (Button)v.findViewById(R.id.off_button);
+
+        switch (mPeripheral.getType()) {
```

Comment:
Should be done in a flexible OOP manner, such as inheritance or a Design Pattern. This switch case is rather ugly and will only give you trouble later in development.